obsah/data_types.py

```diff
@@ -0,0 +1,114 @@
+"""
+obsah data types to be used to validate user input
+"""
+
+import argparse
+import os
+import pathlib
+import re
+import sys
+
+REGISTRY: list[type['BaseType']] = []
+
+
+class BaseType:
+    """
+    Base type class
+    """
+
+    def __init_subclass__(cls, **kwargs):
+        REGISTRY.append(cls)
```

Comment:
Perhaps we should filter this out `RegexType` since it's not directly usable. We could filter out any class that ends with `Type` and treat those as abstract types.